### PR TITLE
[WIP] Deploy Canary Snapshot Controller When Kubernetes Version in Latest

### PR DESCRIPTION
**Pull request by Kartik494**
**What type of PR is this?**

> /kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes-csi/external-snapshotter/issues/516

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: release-tools/prow.sh [hunk 1]
```diff
@@ -838,6 +840,31 @@ install_snapshot_controller() {
       echo "kubectl apply -f $SNAPSHOT_CONTROLLER_YAML"
       kubectl apply -f "$SNAPSHOT_CONTROLLER_YAML"
   fi
+  eval current=${SNAPSHOT_CONTROLLER_YAML}
```
Comment (pohly) [inline release-tools/prow.sh:843]: Why `eval`?

File: release-tools/prow.sh [hunk 2]
```diff
@@ -762,6 +762,8 @@ install_snapshot_crds() {
 
 # Install snapshot controller and associated RBAC, retrying until the pod is running.
 install_snapshot_controller() {
+  TEMP_DIR="$( mktemp -d )"
+  trap 'rm -rf ${TEMP_DIR}' EXIT
   CONTROLLER_DIR="https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/${CSI_SNAPSHOTTER_VERSION}"
   if [[ ${REPO_DIR} == *"external-snapshotter"* ]]; then
```
Comment (pohly) [inline release-tools/prow.sh:768]: Please update both code paths to use `kubectl --kustomize` and the same code. The only difference between them is which image gets used.

File: release-tools/prow.sh [hunk 3]
```diff
@@ -838,6 +840,31 @@ install_snapshot_controller() {
       echo "kubectl apply -f $SNAPSHOT_CONTROLLER_YAML"
       kubectl apply -f "$SNAPSHOT_CONTROLLER_YAML"
   fi
+  eval current=${SNAPSHOT_CONTROLLER_YAML}
+   if [[ CSI_PROW_KUBERNETES_VERSION == "latest" ]]; then
+      run curl "${current}" --output "${TEMP_DIR}"/snapshot_controller.yaml --silent --location
+    else
+        # Even for local files we need to copy because kustomize only supports files inside
+        # the root of a kustomization.
```
Comment (pohly) [inline release-tools/prow.sh:848]: How do you know that we have a local file here? The if check doesn't match.

**Review by pohly — CHANGES_REQUESTED** [review]

File: release-tools/prow.sh (hunk 2 above)
Comment (Kartik494) [reply release-tools/prow.sh:768]: @pohly  Thanks for the suggestions , i will be updating as suggested.

File: release-tools/prow.sh (hunk 3 above)
Comment (Kartik494) [reply release-tools/prow.sh:848]: @pohly  I used the approach here to store the yaml file of external-snapshotter  in a temp dir and then made a kustomization file that will configure and  patch the image file for master to use canary deployment . The implementation for the same is not yet completed will do the same ASAP . 
 `TEMP_DIR="$( mktemp -d )"` 
` trap 'rm -rf ${TEMP_DIR}' EXIT`

**Comment by Kartik494** [comment]
Closing this PR in reference for this  
https://github.com/kubernetes-csi/external-snapshotter/issues/516

**Pull request closed without merging**